### PR TITLE
[Inspector V2] Request the reduced widget tree from Flutter

### PR DESCRIPTION
**Pull request by elliette**
Work towards https://github.com/flutter/devtools/issues/1423

Uses the `fullDetails=false` parameter added in https://github.com/flutter/flutter/pull/157309 to request the smaller widget tree.

### CODE REVIEW

**Review by kenzieschmoll — COMMENTED** [review]
Is there a way to test this to verify we are getting the smaller tree? Perhaps an inspector text golden test that checks this case?

**Comment by elliette** [comment]
> Is there a way to test this to verify we are getting the smaller tree? Perhaps an inspector text golden test that checks this case?

Good point, added a test case to the integration test (which is using a real app). I've added a check that the JSON object for each node only contains the information we expect for the small tree.

File: packages/devtools_app/test/inspector_v2/inspector_integration_test.dart [hunk 1]
```diff
@@ -289,6 +289,47 @@ void main() {
...
+        'createdByLocalProject',
+      ];
+
+      await _loadInspectorUI(tester);
+      final state = tester.state(find.byType(InspectorScreenBody))
+          as InspectorScreenBodyState;
+      final rowsInTree = state.controller.inspectorTree.rowsInTree.value;
+
+      for (final row in rowsInTree) {
+        final detailKeys = row?.node.diagnostic?.json.keys ?? [];
```
[the remark this answers is not in the document]
Comment (elliette) [reply packages/devtools_app/test/inspector_v2/inspector_integration_test.dart:322]: ah good point, done!

File: packages/devtools_app/test/inspector_v2/inspector_integration_test.dart [hunk 2]
```diff
@@ -289,6 +290,59 @@ void main() {
...
+        'stateful',
+      ];
+
+      await _loadInspectorUI(tester);
+      final state = tester.state(find.byType(InspectorScreenBody))
+          as InspectorScreenBodyState;
+      final rowsInTree = state.controller.inspectorTree.rowsInTree.value;
+
+      for (final row in rowsInTree) {
+        final detailKeys = row?.node.diagnostic?.json.keys ?? [];
```
Comment (bkonyi) [inline packages/devtools_app/test/inspector_v2/inspector_integration_test.dart:322]: Ubernit: the default list should be `const <String>[]`

**Review by bkonyi — APPROVED** [review]
LGTM overall!

File: packages/devtools_app/test/inspector_v2/inspector_integration_test.dart (hunk 2 above)
Comment (elliette) [reply packages/devtools_app/test/inspector_v2/inspector_integration_test.dart]: done!

**Pull request merged**